Makefile

```diff
@@ -105,7 +105,8 @@ build-osm-injector: check-go-version clean-osm-injector
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -o ./bin/osm-injector/osm-injector -ldflags "-X $(BUILD_DATE_VAR)=$(BUILD_DATE) -X $(BUILD_VERSION_VAR)=$(VERSION) -X $(BUILD_GITCOMMIT_VAR)=$(GIT_SHA) -s -w" ./cmd/osm-injector
 
 .PHONY: build-osm
-build-osm: check-go-version
+build-osm: CTR_TAG=$(GIT_SHA)
+build-osm: check-go-version $(addprefix docker-push-, init init-osm-controller osm-controller osm-injector)
```

Comment:
Can we avoid docker-push when just building the binary? Since docker-push on the targets can take time, it would increase the time to check if the CLI compiles using `make build-osm`.

Reply:
I'm not sure how we would avoid pushing the images if we want to reference them by default for `osm install`. e.g. If I add a new feature in osm-controller that I want to `osm install`, then a new osm-controller image has to get built at some point before the CLI is invoked.

I suppose I could add a new Makefile target that's CLI+images and have the CLI built before the images so those errors are surfaced first. Then the CLI-only target could be invoked by itself too, but at that point we've basically come full circle back to where we are now.

Overall it seems like this is going to make CLI builds too heavyweight to be worth it, especially since it's solely for convenience and doesn't really solve any problems.